light/client.go

```diff
@@ -1146,3 +1147,25 @@ func (c *Client) providerShouldBeRemoved(err error) bool {
 		errors.As(err, &provider.ErrBadLightBlock{}) ||
 		errors.Is(err, provider.ErrConnectionClosed)
 }
+
+func (c *Client) Status(ctx context.Context) *types.LightClientInfo {
+
+	chunks := make([]string, len(c.witnesses))
+
+	// If primary is in witness list we do not want to count it twice in the number of peers
+	primaryNotInWitnessList := 1
+	for i, val := range c.witnesses {
+		chunks[i] = val.ID()
+		if chunks[i] == c.primary.ID() {
+			primaryNotInWitnessList = 0
+			break
```

Comment:
if we ever hit this condition, we'll end up with a slice that has a number of IDs followed by some number of empty strings, because the slice was allocated with n (zero) values, and you're ending the population/iteration early. 

This might be ok, but it's the kind of thing that can definitely be unexpected for callers of this function. 

I'd definitely either: not specify a length for the slice and let the slice automatically resize as needed (i.e. just use `append()`. It is true that the automatic resizing of slices can be expensive from a `malloc` perspective, but I think that's often overblown, and I think this slice is likely to be small in absolute terms so that's not a huge problem. To avoid this you can pre-allocate the slice, while still creating a slice of 0 length. this would look like `make([]string, 0, len(c.witnesses))`, and then use the `append()` function to add items, safe in the knowledge that the slice will not need to reallocate itself)

Reply:
My bad,  this is a mistake as it is now, we should not break here.  I overlooked that we are filling this array with strings, not just checking whether the ID exists (in which case we would not need an array in the first place). This should be what it was before this last commit (no break statement). Nice catch!